src/write.rs

```diff
@@ -0,0 +1,30 @@
+use std::fmt;
+use Event;
+
+#[derive(Debug)]
+pub struct Error;
+
+impl From<fmt::Error> for Error {
+    fn from(_: fmt::Error) -> Error {
+        Error
+    }
+}
+
+pub trait WriteHtml {
+    fn write_event(&mut self, event: Event) -> Result<(), Error>;
+    fn write_events<'a, I>(&mut self, events: I) -> Result<(), Error>
```

Comment:
```rust
events.fold(Ok(String::new()),
    |mut result, event| result.and_then(
        |result| write!(result, "{}", event).map(|_| result)))

// or

let mut result = String::new();
for event in events {
    try!(write!(result, "{}", event));
}
Ok(result)
```

I think that's simple enough that the `WriteHtml` trait doesn't really give much extra usability.